presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -473,6 +478,23 @@ else if (parameterType.isPrimitive() && !hasNullFlag) {
             }
         }
 
+        private void checkTypeParameters(TypeSignature signature, Method method, ImmutableSet<String> typeParameterNames)
```

Comment:
Add a comment that explains what this function does at high level. For example

    // Check recursively if `signature` contains something like `T<bigint>`